app/src/main/java/ipvc/gymbuddy/app/fragments/ProfileFragment.kt

```diff
@@ -13,4 +15,19 @@ class ProfileFragment : BaseFragment<FragmentProfileBinding>(
         super.onCreate(savedInstanceState)
         viewModel = getViewModel()
     }
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel.user.observe(viewLifecycleOwner, Observer  { user ->
+            user?.let {
+                binding.apply {
+                    name.text = user.name?.toString() ?: ""
+                    firstName.setText(user.name?.toString() ?: "")
+                    gender.setText(user.email?.toString() ?: "")
+                    address.setText(user.address?.toString() ?: "")
+                    role.text = user.role?.name?.toString() ?: ""
+                }
+            }
+        })
+    }
```

Comment:
```suggestion
        viewModel.user.observe(viewLifecycleOwner)  { user ->
                if (user == null) return@observer
                binding.apply {
                    name.text = user.name
                    firstName.text = user.name
                    gender.text = user.email
                    address.text = user.address
                    role.text = user.role?.name
                }
        })
    }
```

also, remove the firstName since we already are showing the user Name on top.